lib/http/node.js

```diff
@@ -123,8 +123,14 @@ function getProxyAgent(proxyOptions, parsedUrl, destination, mock) {
       return mockAgent;
     }
   }
-
-  const agentId = createAgentId(agentOptions.protocol, agentOptions.hostname, agentOptions.keepAlive);
+  let destHost;
```

Comment:
Since we are introducing changes in getProxyAgent method, would it be possible to add test cases covering agent caching bahaviour?

Reply:
See some test cases in TW947

Reply:
Hello @sfc-gh-fpawlowski. I added testing cases here: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/946/files#diff-e251ed9eb74ad284dfcd0fcc9a3d2e34d6b18d4b5cba06f21f040beeb0573013R10. Please take a look when you have a chance. Thank you.

Reply:
Thank You, all looks great now 👍